### DISCUSSION TITLE
Multiserver/service approach

### DISCUSSION THREAD

**Discussion by SilverIce** [azerothcore/azerothcore-wotlk#4841] [General]
Hey. Just curious if there were any attempts to split the core into multiple processes(i.e switch from multithreaded approach)? I heard many times that blizzard went this way

Does it make any sense? Or, with current hardware the performance is no longer a problem?

This is just a discussion, as  I have been and pretty much am out of any development

From what I understand, it's possible to serialize player/or just save to DB, unload and then load by another world process, but the main problem is that there are no process to handle global things like chat, guilds etc

**Comment by FrancescoBorzi** [comment]
hello @SilverIce and welcome. I haven't seen any attempts in doing what you described so far

**Reply by Viste** [reply]
@FrancescoBorzi Laughing Kobold(emu) have this split in 1.12.1 and 2.x.x versions 
They do like blizz: 
auth-node
realm-node
collision(map)-node
world-node
characters-node
blizz also have separate
chat-node
instance-node
bg-node

and they can handle one 1 instance/BG node different realms and etc In short, the usual scaling

**Comment by Yehonal** [comment]
We already use multithread for certain operations inside the core. Maybe you are referring to the possibility of having a better "microservices" architecture, that would be helpful for the crossrealm feature that we don't have yet

**Reply by Viste** [reply]
we can spplite core to more part then auth/world like blizz do this )

**Comment by walkline** [comment]
I'm working on creating a distributed system based on TrinityCore (can be easily adopted with AzerothCore, at least now). Would be glad to get any help with development - https://github.com/walkline/ToCloud9
